### DISCUSSION TITLE
DragDrop

### DISCUSSION THREAD

**Discussion by xiaoliangde** [AvaloniaUI/Avalonia#16544] [Q&A]
### Is your feature request related to a problem? Please describe.

Can Accept DragDrop From OS's Other App(eg. Explorer.   env: avalonia 11.0.6 on windows 11)？

### Describe the solution you'd like

Accept DragDrop From OS's Other App（11.0.6 on windows 11）

**Comment by xiaoliangde** [comment]
class DragSource : IPlatformDragSource
    {
        public unsafe Task<DragDropEffects> DoDragDrop(PointerEventArgs triggerEvent,
            IDataObject data, DragDropEffects allowedEffects)
        {
            Dispatcher.UIThread.VerifyAccess();

            triggerEvent.Pointer.Capture(null);
            
            using var dataObject = new DataObject(data);
            using var src = new OleDragSource();
            var allowed = OleDropTarget.ConvertDropEffect(allowedEffects);
            
            var objPtr = MicroComRuntime.GetNativeIntPtr<Win32Com.IDataObject>(dataObject);
            var srcPtr = MicroComRuntime.GetNativeIntPtr<Win32Com.IDropSource>(src);

            UnmanagedMethods.DoDragDrop(objPtr, srcPtr, (int)allowed, out var finalEffect);
            
            // Force releasing of internal wrapper to avoid memory leak, if drop target keeps com reference.
            dataObject.ReleaseWrapped();

            return Task.FromResult(OleDropTarget.ConvertDropEffect((Win32Com.DropEffect)finalEffect));
        }
    }
   It seems surrport, But Test :
    
            var border = new Border(){Background = Brushes.Aquamarine};

            DragDrop.SetAllowDrop(border, true);
            border.AddHandler(DragDrop.DragEnterEvent, OnDragEnter);
            SetContent(border);
            
            "OnDragEnter" Not Fired. And Inner(Application) Drag Works well.

**Comment by xiaoliangde** [comment]
![image](https://github.com/user-attachments/assets/bd9e2faf-8e74-498a-a279-a2163b116f75)
why not work even by win.adminstrator

**Reply by maxkatz6** [reply]
> why not work even by win.adminstrator

Elevated privilege apps (`Run as Administrator`) can only receive dragndrop events from other apps run as administrator.

**Reply by xiaoliangde** [reply]
> > why not work even by win.adminstrator
> 
> Elevated privilege apps (`Run as Administrator`) can only receive dragndrop events from other apps run as administrator.

I had test any others win(11) by administrator but "DragDrop" not work（with other app's dragdrop).  
wpf or winForm‘s dragdrop works well. 
I'm not familiar with dcom on dragdrop. I used the clipboard to test "DataObject"(with OS's DCom)  and it works well.
